uniffi_bindgen/src/bindings/kotlin/templates/ErrorTemplate.kt

```diff
@@ -78,6 +78,23 @@ internal open class RustError : Structure() {
     }
 }
 
+internal open class InternalError : RustError() {
+    class ByReference: InternalError(), RustErrorReference
+
+    @Suppress("ReturnCount", "TooGenericExceptionThrown", "UNCHECKED_CAST")
+    override fun<E: Exception> intoException(): E {
+        if (!isFailure()) {
+            // It's probably a bad idea to throw here! We're probably leaking something if this is
+            // ever hit! (But we shouldn't ever hit it?)
+            throw RuntimeException("[Bug] intoException called on non-failure!")
+        }
+        val message = this.consumeErrorMessage()
+        return InternalException(message) as E
+    }
+}
+
+class InternalException(message: String) : Exception(message)
```

Comment:
OK, I simplified the error hierarchy here so that we have a catch-all "internal error" instead of getting fancy with reporting panic and invalid handle codes. We might want to report these specially to Sentry, for example, but I totally agree consumers shouldn't be matching on them. WDYT?